### PR TITLE
Update reward and validator selection sections

### PR DESCRIPTION
**Pull request by ilblackdragon**
Fix up reward for accounting the max online threshold.
Adding more information for validator selection.

### CODE REVIEW

File: specs/Economics/README.md [hunk 1]
```diff
@@ -111,23 +110,61 @@ NEAR validators provide their resources in exchange for a reward `epochReward[t]
 
 | Name | Description |
 | - | - |
-| `proposals` | The array of all existing validators, minus the ones which were online less than `ONLINE_THRESHOLD`, plus new validators |
-| `INCLUSION_FEE` | The arbitrary transaction fee that new validators offer to be included in the `proposals`, to mitigate censorship risks by existing validators |
-| `ONLINE_THRESHOLD` | `0.9` |
+| `proposals: Proposal[]` | The array of all new staking transactions that have happened during the epoch |
+| `prev_validators` | The array of all existing validators during the epoch |
 | `epoch[T]` | The epoch when validator[v] is selected from the `proposals` auction array |
```
Comment (mikhailOK) [inline specs/Economics/README.md:115]: There is also the kickout set with slashed validators in the last (2 or 3?) epochs, unstakers and offline, and special tomfoolery when the union of last 2 kickout sets would be everybody and we try not to kickout one (can we assume this case never happens?)

File: specs/Economics/README.md [hunk 2]
```diff
@@ -111,23 +110,61 @@ NEAR validators provide their resources in exchange for a reward `epochReward[t]
...
+    stake: Balance,
+    public_key: PublicKey,
+}
+```
+
+During the epoch, `proposals` are collected, in the form of `Proposal`s.
+At the end of every epoch `T`, next algorithm gets executed to determine validators for epoch `T + 2`:
+
+1. For every validator in `prev_validators` determine `num_blocks_produced`, `num_chunks_produced` based on what they produced during the epoch.
+2.`Remove validators, for whom `(num_blocks_produced / num_blocks_expected + num_chunks_produced / num_chunks_expected) / 2 < ONLINE_THRESHOLD_MIN`.
```
[the remark this answers is not in the document]
Comment (ilblackdragon) [reply specs/Economics/README.md:134]: Apparently no per @bowenwang1996 @SkidanovAlex , added them here.

File: specs/Economics/README.md [hunk 3]
```diff
@@ -139,17 +176,17 @@ Uptime of a specific validator is computed:
...
+    uptime[t][j] = 0.
 ```
 
 Where `expected_produced_blocks` and `expected_produced_chunks` is the number of blocks and chunks respectively that is expected to be produced by given validator `j` in the epoch `t`.
 
 The specific `validator[t][j]` reward for epoch `t` is then proportional to the fraction of stake of this validator from total stake:
 
 ```python
-validator[t][j] = (uptime[t][j] * stake[t][j] * reward[t]) / total_stake[t]
+validatorReward[t][j] = (uptime[t][j] * stake[t][j] * reward[t]) / total_stake[t]
```
[the remark this answers is not in the document]
Comment (ilblackdragon) [reply specs/Economics/README.md:189]: added disclaimer about Rational numbers.

File: specs/Economics/README.md (hunk 1 above)
Comment (ilblackdragon) [reply specs/Economics/README.md:115]: should probably specify this, but let's do this in separate PR. You also have some of this details in EpochManager? Question still stands how these two should be unified.

File: specs/Economics/README.md [hunk 4]
```diff
@@ -16,20 +16,21 @@
...
 | `TREASURY_PCT` | `0.1` |
 | `TREASURY_ACCOUNT_ID` | `treasury` |
 | `CONTRACT_PCT` | `0.3` |
 | `INVALID_STATE_SLASH_PCT` | `0.05` |
 | `ADJ_FEE` | `0.001` |
 | `TOTAL_SEATS` | `100` |
+| `ONLINE_THRESHOLD_MIN` | `0.9` |
+| `ONLINE_THRESHOLD_MAX` | `0.99` |
+| `BLOCK_PRODUCER_KICKOUT_THRESHOLD` | `0.9` |
+| `CHUNK_PRODUCER_KICKOUT_THRESHOLD` | `0.6` |
```
Comment (ilblackdragon) [inline specs/Economics/README.md:33]: @bowenwang1996 @SkidanovAlex feel free to mention here what are these numbers should be at this point.

**Comment by bowenwang1996** [comment]
> Not sure if reward formulas/rounding are supposed to match current code

It will match once https://github.com/nearprotocol/nearcore/pull/2499 is merged

**Comment by ilblackdragon** [comment]
@mikhailOK  please next time LGTM with Approval button 🙆‍♂️

**Pull request merged**